strongbox-web-core/src/main/java/org/carlspring/strongbox/utils/ArtifactControllerHelper.java

```diff
@@ -43,87 +57,133 @@ public static void handlePartialDownload(InputStream is,
+                                                    long inputSize)
     {
-        if (byteRange.getLimit() > 0L && byteRange.getOffset() > 0L)
+        long start = byteRange.getOffset();
+        long end = byteRange.getLimit();
+        long rangeLength = end - start + 1;
+
+        if (end > 0L && start > 0L)
         {
-            logger.debug("Partial content byteRange.getOffset: " + byteRange.getOffset());
-            logger.debug("Partial content byteRange.getLimit: " + byteRange.getLimit());
-            logger.debug("Partial content length: " + (byteRange.getLimit() - byteRange.getOffset()));
+            logger.debug("Partial content byteRange.getOffset: {}", start);
+            logger.debug("Partial content byteRange.getLimit: {}", end);
+            logger.debug("Partial content length: {}", rangeLength);
 
-            return byteRange.getLimit() - byteRange.getOffset();
+            return rangeLength;
         }
-        else if (length > 0L && byteRange.getOffset() > 0L && byteRange.getLimit() == 0L)
+        else if (inputSize > 0L && start > 0L && end == 0L)
         {
-            logger.debug("Partial content length: " + (length - byteRange.getOffset()));
-            return length - byteRange.getOffset();
+            logger.debug("Partial content length: {}", inputSize - start);
+            return inputSize - start;
         }
         else
         {
             return -1L;
         }
     }
 
-    public static void prepareResponseBuilderForPartialRequest(ByteRange br,
-                                                               long length,
-                                                               HttpServletResponse response)
+    private static void prepareResponseBuilderForPartialRequestWithSingleRange(ByteRange byteRange,
+                                                                               long inputLength,
+                                                                               HttpServletResponse response)
     {
-        response.setHeader("Accept-Ranges", "bytes");
-        response.setHeader("Content-Range",
-                           "bytes " + br.getOffset() + "-" + (length - 1L) + "/" + length);
+        String contentRangeHeaderValue = String.format("bytes %d-%d/%d",
+                                                       byteRange.getOffset(),
+                                                       inputLength - 1L,
+                                                       inputLength);
+
+        response.setHeader(HttpHeaders.CONTENT_RANGE, contentRangeHeaderValue);
+
+        response.setHeader(HttpHeaders.PRAGMA, "no-cache");
+
+        response.setStatus(PARTIAL_CONTENT.value());
+    }
+
+    private static void prepareResponseBuilderForPartialRequestWithMultipleRanges(HttpServletResponse response)
+    {
+        response.setContentType("multipart/byteranges; boundary=" + MULTIPART_BOUNDARY);
+
+        response.setHeader(HttpHeaders.PRAGMA, "no-cache");
```

Comment:
is this required ?

Reply:
`Content-Type` is required, as explained [here](https://tools.ietf.org/html/rfc7233#appendix-A).
I'll remove `Pragma`, which it doesn't seem to be needed.